src/main/kotlin/uk/gov/justice/digital/hmpps/deliusapi/service/allocation/AllocationService.kt

```diff
@@ -0,0 +1,184 @@
+   * be the start date of the succeeding one AND any previous one will need it's end date
+   * updating to be the start date of this one.
+   */
+  private fun saveOffenderManager(
+    offender: Offender,
+    newOffenderManager: OffenderManager,
+    activeAtTheTimeOffenderManager: OffenderManager
+  ): OffenderManager {
+
+    newOffenderManager.endDate = activeAtTheTimeOffenderManager.endDate
+    newOffenderManager.active = activeAtTheTimeOffenderManager.active
+    activeAtTheTimeOffenderManager.endDate = newOffenderManager.allocationDate
+    activeAtTheTimeOffenderManager.active = false
+    offenderManagerRepository.save(activeAtTheTimeOffenderManager)
+
+    updateResponsibleOfficer(activeAtTheTimeOffenderManager, newOffenderManager, offender)
+    createInternalTransferContact(offender, newOffenderManager, activeAtTheTimeOffenderManager)
+
+    return offenderManagerRepository.save(newOffenderManager)
+  }
+
+  private fun updateResponsibleOfficer(
+    activeAtTheTimeOffenderManager: OffenderManager,
+    newOffenderManager: OffenderManager,
+    offender: Offender
+  ) {
+    // update the existing RO
+    val activeAtTheTimeResponsibleOfficer = responsibleOfficerRepository
+      .findByOffenderManagerId(activeAtTheTimeOffenderManager.id!!, newOffenderManager.allocationDate)
+      ?: throw BadRequestException("Invalid allocation date - couldn't retrieve responsible officer")
+
+    val newResponsibleOfficer = ResponsibleOfficer(
+      offender = offender,
+      startDate = newOffenderManager.allocationDate,
+      endDate = activeAtTheTimeResponsibleOfficer.endDate,
+      communityOffenderManager = newOffenderManager
+    )
+    responsibleOfficerRepository.save(newResponsibleOfficer)
+    activeAtTheTimeResponsibleOfficer.endDate = newOffenderManager.allocationDate
+    responsibleOfficerRepository.save(activeAtTheTimeResponsibleOfficer)
+    // TODO contact type ROC
+  }
+
+  private fun createInternalTransferContact(offender: Offender, newOffenderManager: OffenderManager, oldManager: OffenderManager) {
+    systemContactService.createSystemContact(
+      NewSystemContact(
+        type = WellKnownContactType.INTERNAL_TRANSFER,
+        offenderId = offender.id,
+        date = newOffenderManager.allocationDate.toLocalDate(),
+        teamId = newOffenderManager.team.id,
+        staffId = newOffenderManager.staffAllocation.id,
+        providerId = newOffenderManager.provider.id,
+        notes = listOfNotNull(
+          "Transfer Reason: Internal Transfer",
+          "Transfer Date: ${newOffenderManager.allocationDate.format(DateTimeFormatter.ofPattern("dd/MM/yyyy"))}",
+          "From Trust: ${newOffenderManager.provider.description}",
+          "From Team: ${oldManager.team.description}",
+          "From Officer: ${oldManager.staffAllocation.getDisplayName()}",
+          "-------------------------------",
+          "", // <- ends in an empty newline
+        ).joinToString("\n"),
```

Comment:
This could be a multi-line string.  `listOfNotNull` seems redundant as nothing in the list can be null.
```suggestion
        notes = """
          Transfer Reason: Internal Transfer
          Transfer Date: ${newOffenderManager.allocationDate.format(DateTimeFormatter.ofPattern("dd/MM/yyyy"))}
          From Trust: ${newOffenderManager.provider.description}
          From Team: ${oldManager.team.description}
          From Officer: ${oldManager.staffAllocation.getDisplayName()}
          -------------------------------
          
        """.trimIndent(),
```